### PR TITLE
Feat/component lifecycle

### PR DESCRIPTION
**Pull request by ex3emxl**

### CODE REVIEW

File: src/components/clock/Clock.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+class Clock extends Component {
+    state = {
+        date: new Date()
+    };
+
+    componentDidMount() {
+        this.timerID = setInterval(
+            () => this.tick(),
```
Comment (iliatcymbal) [inline src/components/clock/Clock.js:10]: no need extra function here `() => ...`

File: src/components/clock/Clock.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    tick() {
+        this.setState({
+            date: new Date()
+        });
+    }
+
+    render() {
+        const { date } = this.state;
+        return (<div className='date'>
```
Comment (iliatcymbal) [inline src/components/clock/Clock.js]: put jsx from new line

File: src/components/clock/Clock.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <p>{date.toLocaleDateString()}</p>
+            <p>{date.toLocaleTimeString()}</p>
+        </div>);
```
Comment (iliatcymbal) [inline src/components/clock/Clock.js]: put bracket on new line

File: src/components/content/Content.js [hunk 4]
```diff
@@ -18,10 +20,10 @@ const Content = () => (
             name="Se Mi"
         />
         <br/>
-
+        <Date/>
         <Counter/>
         <br/>
-        <Numbers from=1 to=7 even/>
+        <Numbers from={1} to={7} even/>
```
Comment (iliatcymbal) [inline src/components/content/Content.js]: space before `/>`

File: src/components/content/Content.js [hunk 5]
```diff
@@ -18,10 +20,10 @@ const Content = () => (
             name="Se Mi"
         />
         <br/>
-
+        <Date/>
```
Comment (iliatcymbal) [inline src/components/content/Content.js]: space before `/>`

File: src/components/content/Content.js [hunk 6]
```diff
@@ -30,7 +32,8 @@ const Content = () => (
         <br/>
         <Button/>
         <br/>
-        <Show />
+        <Show/>
```
Comment (iliatcymbal) [inline src/components/content/Content.js]: space before `/>`

File: src/components/content/Content.js [hunk 7]
```diff
@@ -30,7 +32,8 @@ const Content = () => (
...
+        <Scintillation/>
```
Comment (iliatcymbal) [inline src/components/content/Content.js]: space before `/>`

File: src/components/mount/mount.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import './mount.scss';
+
+class Mount extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            status: 'init'
+        };
+
+        this.updateStatus = this.updateStatus.bind(this);
```
Comment (iliatcymbal) [inline src/components/mount/mount.js]: do not use bind, use arrow function to save context

File: src/components/scintillation/Scintillation.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+        return (
+            <div>
+                <button onClick={() => this.setState({counter: counter + 1})}>Inc</button>
+                <span className="count-text">{this.state.counter}</span>
+                <br/><br/>
+                <button onClick={() => this.setState({show: !show})}>
+                    {show ? 'Destroy' : 'Create'}
+                </button>
+                {show && <Mount code={counter} color={this.state.color}/>}
```
Comment (iliatcymbal) [inline src/components/scintillation/Scintillation.js]: color logic should be defined inside `Mount` component, you shouldn't pass it via props.

File: src/components/scintillation/Scintillation.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            this.setState({color: 'green'});
+        }
+        if (
+            this.state.counter !== prevState.counter &&
+            this.state.counter % 5 === 0
+        ) {
+            this.setState({color: 'blue'});
+        }
+        if (
+            this.state.counter !== prevState.counter &&
```
Comment (iliatcymbal) [inline src/components/scintillation/Scintillation.js]: you have this condition in 4 places. Why don't you put it once in the beginning?
`if (this.state.counter !== prevState.counter) return;`

**Review by iliatcymbal — CHANGES_REQUESTED** [review]
Please, see my comments

File: src/components/clock/Clock.js (hunk 1 above)
Comment (ex3emxl) [reply src/components/clock/Clock.js:10]: It's not work without extra function

Comment (iliatcymbal) [reply src/components/clock/Clock.js:10]: really? What an error?

**Review by iliatcymbal — APPROVED** [review]
Looks much better, but still need discuss this: 
```
componentDidMount() {
        this.timerID = setInterval(
            () => this.tick(),
```

**Pull request merged**